src/Generator/Generators/CLI/CLIHeaders.cs

```diff
@@ -745,31 +745,27 @@ public bool GenerateTypedef(TypedefNameDecl typedef)
             if (!typedef.IsGenerated)
                 return false;
 
-            FunctionType function;
-            if (typedef.Type.IsPointerTo(out function))
+            var functionType = typedef.Type as FunctionType;
+            if (typedef.Type.IsPointerTo(out functionType) || functionType != null)
             {
                 PushBlock(BlockKind.Typedef, typedef);
                 GenerateDeclarationCommon(typedef);
 
                 var insideClass = typedef.Namespace is Class;
 
                 var attributedType = typedef.Type.GetPointee() as AttributedType;
```

Comment:
I know this is old code but I think we now check for attributed types when desugaring so just desugar instead.